mlflow/models/docker_utils.py

```diff
@@ -55,25 +56,34 @@ def _get_maven_proxy():
     assert parsed_https_proxy.hostname is not None, "Invalid `https_proxy` hostname."
     assert parsed_https_proxy.port is not None, f"Invalid proxy port: {parsed_https_proxy.port}"
 
-    maven_proxy_options = (
+    non_proxy_hosts = '|'.join(no_proxy.split(','))
+
+    maven_proxy_options = [
         "-DproxySet=true",
         f"-Dhttp.proxyHost={parsed_http_proxy.hostname}",
         f"-Dhttp.proxyPort={parsed_http_proxy.port}",
+        f"-Dhttp.nonProxyHosts={non_proxy_hosts}",
         f"-Dhttps.proxyHost={parsed_https_proxy.hostname}",
         f"-Dhttps.proxyPort={parsed_https_proxy.port}",
-        "-Dhttps.nonProxyHosts=repo.maven.apache.org",
-    )
+        f"-Dhttps.nonProxyHosts={non_proxy_hosts}",
+    ]
 
     if parsed_http_proxy.username is None or parsed_http_proxy.password is None:
-        return " ".join(maven_proxy_options)
-
-    return " ".join(
-        (
-            *maven_proxy_options,
+        maven_proxy_options.extend([
             f"-Dhttp.proxyUser={parsed_http_proxy.username}",
             f"-Dhttp.proxyPassword={parsed_http_proxy.password}",
-        )
-    )
+        ])
+    if parsed_https_proxy.username is None or parsed_https_proxy.password is None:
+        maven_proxy_options.extend([
+            f"-Dhttps.proxyUser={parsed_https_proxy.username}",
+            f"-Dhttps.proxyPassword={parsed_https_proxy.password}",
```

Comment:
if one of them is None, shall we skip adding the property ?

Reply:
Yeah, this is just plain backwards, which I discovered when testing it in my "in-the-wild" environment today (and why I left this as draft). Unfortunately, its a moot point. Maven does not actually respect any of those command line flags, it *requires* the creation of an XML file for authenticated proxies. This isn't too difficult in concept (just create the file, run maven, then delete it, all in the same RUN), but its tedious. As well, an even better idea would be to actually parse the proxy variables passed via --build-arg, rather than hard-coding them into the dockerfile, which would make it so that this part of the dockerfile could be made static, as well as remove any concerns with comitting those proxy credentials into git. I think a complete rewrite of the maven proxy handling is in order, which I'll be working on tomorrow.